audits/internal3/README.md

```diff
@@ -0,0 +1,81 @@
+# meme-ooorr
+The review has been performed based on the contract code in the following repository:<br>
+`https://github.com/dvilelaf/meme-ooorr` <br>
+commit: `9b427618db7bd233651476de9abe12c18fbd236a` or `v0.2.0-internal-audit2` <br>
+
+## Objectives
+The audit focused on Meme* contracts <BR>
+
+### Flatten version
+Flatten version of contracts. [contracts](https://github.com/dvilelaf/meme-ooorr/blob/main/audits/internal3/analysis/contracts)
+
+### Security issues
+Details in [slither_full](https://github.com/dvilelaf/meme-ooorr/blob/main/audits/internal3/analysis/slither_full.txt) <br>
+All false positive.
+
+## Issues
+### Medium? Critical?: to discussion, sandwich attack
+```
+attaсker contract => function unleashThisMeme => _createThisMeme => 
+bytes32 randomNonce = keccak256(abi.encodePacked(block.timestamp, block.prevrandao, msg.sender, memeNonce));
+        randomNonce = keccak256(abi.encodePacked(randomNonce));
+Can be calculated in the attacking contract (because deterministic in same block)
+Can predict: memeToken := create2(0x0, add(0x20, payload), mload(payload), memeNonce)
+Call before own contract:
+        // Calculate the price ratio (amount1 / amount0) scaled by 1e18 to avoid floating point issues
+        uint256 priceX96 = (amount1 * 1e18) / amount0;
+
+        // Calculate the square root of the price ratio in X96 format
+        uint160 sqrtPriceX96 = uint160((FixedPointMathLib.sqrt(priceX96) * (1 << 96)) / 1e9);
+
+        // Create a pool
+        IUniswapV3(uniV3PositionManager).createAndInitializePoolIfNecessary(token0, token1, FEE_TIER, sqrtPriceX96);
+So, we must revert if pool exist at moment unleashThisMeme        
```

Comment:
Isn't it automatically reverting because the pool initialisation will fail if its not identical? If it is identical there's nothing to worry?

Reply:
I think this can be done before our contract is called.

/// @title Creates and initializes V3 Pools
abstract contract PoolInitializer is IPoolInitializer, PeripheryImmutableState {
    /// @inheritdoc IPoolInitializer
    function createAndInitializePoolIfNecessary(
        address token0,
        address token1,
        uint24 fee,
        uint160 sqrtPriceX96
    ) external payable override returns (address pool) {
        require(token0 < token1);
        pool = IUniswapV3Factory(factory).getPool(token0, token1, fee); << call this before own contract

        if (pool == address(0)) { << own contract skip this
            pool = IUniswapV3Factory(factory).createPool(token0, token1, fee);
            IUniswapV3Pool(pool).initialize(sqrtPriceX96);
        } else {
            (uint160 sqrtPriceX96Existing, , , , , , ) = IUniswapV3Pool(pool).slot0();
            if (sqrtPriceX96Existing == 0) { << own contract skip this
                IUniswapV3Pool(pool).initialize(sqrtPriceX96);
            }
        }
    }
}

Reply:
pool = IUniswapV3Factory(factory).createPool(token0, token1, fee); Does not require tokens.

Reply:
I am fine with reverting if the pool is already created. And no, it is not automatically reverting.